### PR TITLE
Use in-memory db for rest api tests

### PR DESCRIPTION
**Pull request by dnewh**
This updates the rest api tests to use an in-memory sqlite db. This
offers a couple benefits. One, this "resets" the database between tests
so that the db does not need to be cleared manually. Two, this does not
create any database files that would need to be manually cleaned up
after the tests.

Signed-off-by: Davey Newhall <newhall@bitwise.io>

### CODE REVIEW

**Review by peterschwarz — COMMENTED** [review]
Make sure it is added to any dockerignore files

**Review by KevinAnsleyJohnson — DISMISSED** [review]
LGTM

**Review by vaporos — CHANGES_REQUESTED** [review]
The test should be modified to create this in a temporary directory and then clean it up after.

**Review by vaporos — APPROVED** [review]

File: daemon/src/rest_api/mod.rs [hunk 1]
```diff
@@ -767,25 +742,12 @@ mod test {
...
-                srv.url(&format!("/agent?service_id={}", TEST_SERVICE_ID)),
-            )
-            .send()
-            .await
-            .unwrap();
-        assert!(response.status().is_success());
-        let body: AgentListSlice =
-            serde_json::from_slice(&*response.body().await.unwrap()).unwrap();
-        assert!(body.data.is_empty());
+        // let pool = create_connection_pool_and_migrate();
```
Comment (dplumb94) [inline daemon/src/rest_api/mod.rs:745]: remove commented code

**Review by dplumb94 — CHANGES_REQUESTED** [review]

File: daemon/src/rest_api/mod.rs (hunk 1 above)
Comment (dnewh) [reply daemon/src/rest_api/mod.rs]: fixed

**Review by dplumb94 — APPROVED** [review]
Worked locally, LGTM

**Review by vaporos — APPROVED** [review]

**Pull request merged**